core/src/fields.rs

```diff
@@ -542,23 +550,66 @@ impl MongoFields {
                             // The collection does not match the filter, moving to the next one
                             continue;
                         }
-                        let get_schema_cmd = doc! {"sqlGetSchema": collection_name.clone()};
 
                         let db = mongo_connection.client.database(&self.current_db_name);
-                        let current_col_metadata_response: Result<SqlGetSchemaResponse> =
-                            mongodb::bson::from_document(
-                                db.run_command(get_schema_cmd).await.unwrap(),
-                            )
-                            .map_err(|e| {
-                                Error::CollectionDeserialization(collection_name.clone(), e)
-                            });
-                        if let Err(error) = current_col_metadata_response {
-                            // If there is an Error while deserializing the schema, we won't show any columns for it
-                            warnings.push(error);
-                            continue;
-                        }
-                        let current_col_metadata_response: ResultSetSchema =
-                            current_col_metadata_response.unwrap().into();
+
+                        let current_col_metadata_response: ResultSetSchema = if mongo_connection
+                            .cluster_type
+                            == MongoClusterType::AtlasDataFederation
+                        {
+                            let get_schema_cmd = doc! {"sqlGetSchema": &collection_name};
+
+                            let sql_get_schema_response: Result<SqlGetSchemaResponse> =
+                                mongodb::bson::from_document(
+                                    db.run_command(get_schema_cmd).await.unwrap(),
+                                )
+                                .map_err(|e| {
+                                    Error::CollectionDeserialization(collection_name.clone(), e)
```

Comment:
How about an integration test that confirms we get this on a deserialization error?

Reply:
To get this error, there would have to be a document in the `__sql_schemas` collection that has the `schema` field set to either the wrong type (an int) or an incorrectly made schema that can't be correctly deserialized. Are you sure we need an integration test for this? It's a very specific case. I think maybe a unit test would work better?

Reply:
I see. A unit test would be fine then.